src/apps/apploader/app.js

```diff
@@ -85,77 +86,85 @@ define(function(require) {
+				isVisible = _.get(self.appFlags, 'modal.isVisible', function() { return false; })();
+
+			self._destroy();
+
+			if (isVisible) {
+				self.render();
+			}
+		},
+
 		/**
 		 * @param  {Function} [pArgs.callback]
 		 */
 		render: function(pArgs) {
 			var self = this,
-				globalCallback = _.get(pArgs, 'callback');
+				callback = _.get(pArgs, 'callback', function() {});
 
-			monster.waterfall([
-				function shouldRender(callback) {
-					monster.pub('myaccount.hasToShowWalkthrough', function(hasToShowWalkthrough) {
-						callback(hasToShowWalkthrough ? 'doNotRender' : null);
-					});
-				},
-				function maybeFetchApps(callback) {
-					if (self.isRendered()) {
-						return callback('isRendered');
-					}
-					self.getUserApps(_.partial(callback, null));
-				},
-				function insertAppLinks(appList, callback) {
-					callback(null, self.insertAppLinks(appList));
-				},
-				function renderTemplate(appList, callback) {
-					var templateTypes = {
-							dropdown: {
-								name: 'appList',
-								defaultApp: _.head(appList),
-								insert: function(template) {
-									self.bindDropdownApploaderEvents(template);
-
-									$('#appList').empty().append(template);
-								}
-							},
-							modal: {
-								name: 'app',
-								defaultApp: monster.ui.formatIconApp(appList[0]),
-								insert: function(template) {
-									self.bindEvents(template, appList);
-
-									self.appFlags.modal = monster.ui.fullScreenModal(template, {
-										hideClose: true,
-										destroyOnClose: false
-									});
-								}
-							}
-						},
-						templateData = _.get(templateTypes, monster.config.whitelabel.useDropdownApploader ? 'dropdown' : 'modal'),
-						$template = $(self.getTemplate({
-							name: templateData.name,
-							data: _.merge({
-								allowAppstore: monster.apps.auth.currentUser.priv_level === 'admin' && !monster.config.whitelabel.hideAppStore,
-								apps: appList
-							}, _.pick(templateData, [
-								'defaultApp'
-							]))
-						}));
-
-					templateData.insert($template);
-
-					callback(null);
+			monster.series([
+				function shouldRender(next) {
+					monster.pub('myaccount.hasToShowWalkthrough', next);
 				}
-			], function(exitReason) {
-				if (exitReason === 'doNotRender') {
+			], function(err) {
+				if (err) {
 					return;
 				}
-				if (exitReason === 'isRendered') {
+				if (self.isRendered()) {
 					return self.show({
-						callback: globalCallback
+						callback: callback
 					});
 				}
-				globalCallback && globalCallback();
+				var defaultApp = monster.util.getCurrentUserDefaultApp(),
+					actionsPerId = _
+						.chain([
+							monster.util.listAppStoreMetadata('user'),
+							monster.util.listAppLinks()
+						])
+						.flatten()
+						.keyBy('id')
+						.value(),
+					appList = _
+						.chain(_.get(monster.apps, 'auth.currentUser.appList', []))
+						.map(_.partial(_.ary(_.get, 2), actionsPerId))
+						.reject(_.isUndefined)
+						.value(),
```

Comment:
Should the app links also be listed as part of the user apps? It seems that it was not required in the previous implementation.

Reply:
If you look closely, they were indeed [insert ](https://github.com/2600hz/monster-ui/blob/master/src/apps/apploader/app.js#L108)at an earlier point, and we definitely want to display them to the end user.

Reply:
Sorry, what I meant is that the app links are not displayed in the app menu in this branch because they are filtered out for not being present in `monster.apps.auth.currentUser.appList`. I have tried this setup in config.js:
```javascript
whitelabel: {
	// ...
	appLinks: {
		'http://www.google.com': {
			icon: 'https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg',
			i18n: {
				'en-US': {
					label: 'Google',
					description: 'Google home page'
				}
			}
		}
	}
}
```
And the app link is listed in the `master` branch, but not in `MSTR-197`.

Reply:
Good call, with the new events triggering current appstore/user refresh, fetching apps metadata needs to happen AFTER authenticated entities are retrieved, as that's when the app list is potentially updated/noramlized.